### ISSUE TITLE
Boolean operations on non-boolean literals should cause compiler warnings

### ISSUE THREAD

**Issue by ajpaon** [google/closure-compiler#1290]
I accidentally wrote a bit of code that looks like this:

var myString = 'string1' || 'string2';

This ought to be caught at compile time, as myString will never have a value of 'string2'.

**Comment by blickly** [comment]
As long as the first string's value isn't '', that is.

**Comment by ajpaon** [comment]
Good point - but then the variable will always be set to the second string. Either way there's dead code that the compiler can call attention to.

**Comment by Dominator008** [comment]
@ajpaon The compiler already optimizes this. Using `SIMPLE_OPTIMIZATIONS`:
```js
var myString = 'string1' || 'string2';
```
becomes
```
var myString = "string1";
```
and
```js
var myString = '' || 'string2';
```
becomes
```js
var myString = "string2";
```
It might be good to have a lint check for this though.
